MainRepository/Program.cs

```diff
@@ -1,7 +1,215 @@
+                  DisplayBoard (word);
+                  while (true) {
+                     var lastKey = Console.ReadKey (true);
+                     if (lastKey.Key == ConsoleKey.Enter && ValidWord ()) {
+                        Console.WriteLine (new string (' ', Console.WindowWidth));
+                        break;
+                     }
+                     if (lastKey.Key == ConsoleKey.Enter && !ValidWord ()) {
+                        mBadWord = string.Join ("", mList).ToUpper ();
+                        var cursorTop = mIsInvalid ? Console.CursorTop - 1 : Console.CursorTop;
+                        Console.SetCursorPosition (mHalfWidth - 1, cursorTop);
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        Console.WriteLine ($"{mBadWord} is not a word");
+                        Console.ResetColor ();
+                        mIsInvalid = true;
+                     }
+                     if (lastKey.Key == ConsoleKey.Backspace) {
+                        mWordle[i, 4] = mCircle;
+                        mWordle[i + 1, 0] = mDot;
+                        j--;
+                        mList.RemoveAt (4);
+                        Console.SetCursorPosition (mHalfWidth - 1, Console.CursorTop - 1);
+                        mIsInvalid = false;
+                        Console.WriteLine (new string (' ', Console.WindowWidth));
+                        break;
+                     }
+                  }
+               }
+               if (i < 5 && j == 4) mWordle[i + 1, 0] = mCircle;
+               DisplayBoard (word);
+               if (mList.Count == 5) {
+                  DisplayBoard (word, true);
+                  if (mGreenList.Count == 5) {
+                     mGameOver = true;
+                     mCount++;
+                     return;
+                  }
+                  if (!ValidWord ()) j--;
+                  else mList.Clear ();
+               }
+            }
+         }
+         mCount++;
+      }
+   }
+
+   /// <summary>Sets the foreground color for the alphabet</summary>
+   public char SetForegroundClr (char ch) {
+      if (mGreenList.Contains (ch)) Console.ForegroundColor = ConsoleColor.Green;
+      else if (mBlueList.Contains (ch))
+         Console.ForegroundColor = ConsoleColor.Blue;
+      else if (mGrayList.Contains (ch))
+         Console.ForegroundColor = ConsoleColor.DarkGray;
+      else Console.ForegroundColor = ConsoleColor.Gray;
+      return ch;
+   }
+
+   /// <summary>Displays the current state of the board</summary>
+   public void DisplayBoard (string word, bool isEntered = false) {
+      Console.CursorLeft = 0;
+      Console.CursorTop = 0;
+      Console.CursorVisible = false;
+      Console.WriteLine ();
+      Console.CursorLeft = mHalfWidth - 5;
+      for (int i = 0; i < 6; i++) {
+         for (int j = 0; j < 5; j++) {
+            if (i < mRow || isEntered) {
+               if (word[j] == mWordle[i, j])

```

Comment:
```suggestion
               if (word[j] == **mWordle[i, j]**)

```
This is accessed multiple times, instead save it in a local variable and use it